### PR TITLE
Add some debug tips for debugging peons

### PR DESCRIPTION
**Pull request by maytasm**
Add some debug tips for debugging peons

### Description

- Add some debug tips for debugging peons
- Enable (open debug ports) for peons on Integration tests Druid cluster by default

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by suneet-s — APPROVED** [review]
Nice tip! 🚀

File: dev/intellij-setup.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ Before running or debugging the apps, you should do a `mvn clean install -Pdist
 
 You may also add `-Ddruid.console.skip=true` to the command if you're focusing on backend servers instead of frontend project. This option saves great building time.
 
+## Debugging Running Druid from Intellij
+Intellij IDEA debugger can attach to a local or remote Java process (Druid process). 
+First, you must enable debugging in the application.
```
Comment (kfaraz) [inline dev/intellij-setup.md:74]: Would it be better to write these steps in a list format?

Something like:
```suggestion
Follow these steps to debug a Druid process using the IntelliJ IDEA debugger.
1. Enable debugging in the Druid process
   a. For Druid services (Overlord, Coordinator, etc.) ...
   b. For peons (...
2. Find the port assigned to the Druid process.
   a. For Druid services, you can find it in ...
   b. For peons, you can find the assigned ephemeral port ... task log
3. Create a remote configuration in the Run/Debug configurations dialog ...
4. Start (debug) the above remote configuration
```

**Review by kfaraz — COMMENTED** [review]
Overall LGTM. Added minor suggestions.

File: dev/intellij-setup.md [hunk 2]
```diff
@@ -69,6 +69,14 @@ Before running or debugging the apps, you should do a `mvn clean install -Pdist
 
 You may also add `-Ddruid.console.skip=true` to the command if you're focusing on backend servers instead of frontend project. This option saves great building time.
 
+## Debugging Running Druid from Intellij
```
[the remark this answers is not in the document]
Comment (maytasm) [reply dev/intellij-setup.md:72]: Done

File: integration-tests/README.md [hunk 3]
```diff
@@ -212,6 +212,7 @@ For your convenience, Druid processes running inside Docker have been debugging
 | Historical | 5007 |
 | Middlemanager | 5008 |
 | Overlord | 5009 |
+| Peons (Workers on Middle Manager) |  Ephemeral port assigned by debugger (check task log for port assigned to each task)
```
[the remark this answers is not in the document]
Comment (maytasm) [reply integration-tests/README.md:215]: Done

File: dev/intellij-setup.md (hunk 1 above)
Comment (maytasm) [reply dev/intellij-setup.md:74]: Done

**Pull request merged**